x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/exporter/http/HttpExporter.java

```diff
@@ -183,7 +183,45 @@ public void validate(final List<String> hosts, final Map<Setting<?>, Object> set
      */
     public static final Setting.AffixSetting<String> AUTH_PASSWORD_SETTING =
             Setting.affixKeySetting("xpack.monitoring.exporters.","auth.password",
-                    (key) -> Setting.simpleString(key, Property.Dynamic, Property.NodeScope, Property.Filtered));
+                    (key) -> Setting.simpleString(key,
+                        new Setting.Validator<String>() {
+                            @Override
+                            public void validate(String password) {
+                                // no password validation that is independent of other settings
+                            }
+
+                            @Override
+                            public void validate(String password, Map<Setting<?>, Object> settings) {
```

Comment:
Don't we need similar validation in AUTH_USERNAME_SETTING? Otherwise we could have username exist but no password?

Reply:
@rjernst, that validation was added in https://github.com/elastic/elasticsearch/pull/47821.

Reply:
Ah ok, thanks. i guess this PR is out of date with master?